docs/website/docs/general-usage/http/rest-client.md

```diff
@@ -494,6 +523,24 @@ class CustomAuth(AuthConfigBase):
         return request
 ```
 
+**Custom combined auth:**
+Sometimes you need to pass authentication parameters via headers as well as query params
+
+```py
+from dlt.sources.helpers.rest_client.auth import AuthConfigBase
+
+class CombinedAuth(AuthConfigBase):
+    def __init__(self, client_id: str, client_secret: str):
+        self.client_id = client_id
+        self.client_secret = client_secret
+
+    def __call__(self, request):
+        # Modify the request object to include the necessary authentication headers and request params
+        request.headers["Authorization"] = f"Bearer {self.client_secret}"
+        request.prepare_url(request.url, {"client_id": self.client_id})
+        return request
+```
```

Comment:
I'm not convinced we need this example here: the difference that I see here from the previous example is that it shows that `request` is a PreparedRequest instance (because we can call `prepare_url` on it). I think showing that `request` is a PreparedRequest has a value, so instead of the example I would add explicit types in the previous example (like you did with token: str) and add a text description elaborating on what `__call__` actually is receiving and linking Requests docs from PreparedRequest so the reader can quickly see what's possible.